session_factory.go

```diff
@@ -431,6 +431,18 @@ func (f sessionFactory) newSession(
 		s.DisableMessagePersist = !persistMessages
 	}
 
+	if settings.HasSetting(config.InChanCapacity) {
+		if s.InChanCapacity, err = settings.IntSetting(config.InChanCapacity); err != nil {
+			return
+		} else if s.InChanCapacity < 0 {
+			err = IncorrectFormatForSetting{Setting: config.InChanCapacity, Value: []byte(strconv.Itoa(s.InChanCapacity))}
+			return
+		}
+	} else {
+		// Default to 1 buffered message per channel
+		s.InChanCapacity = 1
```

Comment:
This changes the existing behavior to buffered channel from unbuffered. For those who reset the session on disconnect, there is edge case where they lose the last message in case the application crashes.

Reply:
Maybe we can add a warning when the new release is published